internal/util/util.go

```diff
@@ -481,15 +480,20 @@ func GetFacet(s *pb.SourceSeries) *pb.Facet {
 // GetFacetID retrieves a hash string for a given protobuf message.
 // Note this should be restrict to a request scope.
 func GetFacetID(m *pb.Facet) string {
-	h := fnv.New32a()
-	_, _ = h.Write([]byte(strings.Join([]string{
+	// Only include fields that are set in hash.
+	// This is so the hashes stay consistent if more fields are added.
+	s := strings.Join([]string{
 		m.ImportName,
 		m.MeasurementMethod,
 		m.ObservationPeriod,
 		m.ScalingFactor,
 		m.Unit,
-		strconv.FormatBool(m.IsDcAggregate),
-	}, "-")))
+	}, "")
```

Comment:
Any reason to not continue joining them with a "-"? If you leave it that way, majority of the goldens would likely stay the same.

Reply:
This is so if we add more string fields the hash won't change (if we use a connector, then there will be a connector even if the field is an empty string). 

If we want to preserve the existing hashes, I could keep the connector but for any future fields just append if it exists, rather than adding to the join (like for the IsDcAggregate)

Reply:
I think without a separator, we may run into bugs in some scenarios. e.g. imagine a facet with Unit = foo and everything else not set and another one with MeasurementMethod = foo and everything else not set. Without a separator, both would produce the same string and hence yield the same facet ID.

Reply:
Makes sense, I added the "-" back 

(The ids are still changing in this PR because we're making the is_dc_aggregate optional, while in the previous PR we always added it)